plotly/figure_factory/_facet_grid.py

```diff
@@ -0,0 +1,1073 @@
+                                facet_col_labels, trace_type, size,
+                                flipped_rows, flipped_cols, show_boxes,
+                                kwargs, kwargs_marker):
+
+    fig = make_subplots(rows=num_of_rows, cols=num_of_cols,
+                        shared_xaxes=True, shared_yaxes=True,
+                        horizontal_spacing=SUBPLOT_SPACING,
+                        vertical_spacing=SUBPLOT_SPACING, print_grid=False)
+
+    annotations = []
+    if not facet_row and not facet_col:
+        trace = graph_objs.Scatter(
+            x=df[x].tolist(),
+            y=df[y].tolist(),
+            mode='markers',
+            type=trace_type,
+            marker=dict(
+                color=df[color_name],
+                size=size,
+                colorscale=colormap,
+                showscale=True,
+                **kwargs_marker
+            ),
+            **kwargs
+        )
+        fig.append_trace(trace, 1, 1)
+
+    elif facet_row and not facet_col:
+        groups_by_facet_row = list(df.groupby(facet_row))
+        for j, group in enumerate(groups_by_facet_row):
+            trace = graph_objs.Scatter(
+                x=group[1][x].tolist(),
+                y=group[1][y].tolist(),
+                mode='markers',
+                type=trace_type,
+                marker=dict(
+                    color=df[color_name].tolist(),
+                    size=size,
+                    colorscale=colormap,
+                    showscale=True,
+                    colorbar=dict(x=1.15),
+                    **kwargs_marker
+                ),
+                **kwargs
+            )
+            fig.append_trace(trace, j + 1, 1)
+
+            label = _return_label(group[0], facet_row_labels, facet_row)
+
+            annotations.append(
+                _annotation_dict(label, num_of_rows - j, num_of_rows,
+                                 row_col='row', flipped=flipped_rows)
+            )
+
+    elif not facet_row and facet_col:
+        groups_by_facet_col = list(df.groupby(facet_col))
+        for j, group in enumerate(groups_by_facet_col):
+            trace = graph_objs.Scatter(
+                x=group[1][x].tolist(),
+                y=group[1][y].tolist(),
+                mode='markers',
+                type=trace_type,
+                marker=dict(
+                    color=df[color_name].tolist(),
+                    size=size,
+                    colorscale=colormap,
+                    showscale=True,
+                    colorbar=dict(x=1.15),
+                    **kwargs_marker
+                ),
+                **kwargs
+            )
+            fig.append_trace(trace, 1, j + 1)
+
+            label = _return_label(group[0], facet_col_labels, facet_col)
+
+            annotations.append(
+                _annotation_dict(label, j + 1, num_of_cols, row_col='col',
+                                 flipped=flipped_cols)
+            )
```

Comment:
these two blocks could be a little drier, but it's not a big deal. Something like this:
```
if (facet_row and not facet_col) or (not facet_row and facet_col):
    groups = list(df.groupby(facet_row if facet_row else facet_col))
    for j, group in enumerate(groups):
        ...
        fig.append_trace(trace, j + 1 if facet_row else 1, 1 if facet_row else j +1)
    ... 
```